pypsa/plot.py

```diff
@@ -1233,3 +1243,160 @@ def iplot(
+        n.buses, geometry=gpd.points_from_xy(n.buses.x, n.buses.y), crs=crs
+    )
+
+    n.links = n.links[n.links.carrier == "DC"]  # Filter out non DC links
+    gdf_links = gpd.GeoDataFrame(
+        n.links,
+        geometry=[
+            LineString(
+                [
+                    (n.buses.loc[link.bus0, "x"], n.buses.loc[link.bus0, "y"]),
+                    (n.buses.loc[link.bus1, "x"], n.buses.loc[link.bus1, "y"]),
+                ]
+            )
+            for link in n.links.itertuples()
+        ],
+        crs="EPSG:4326",
+    )
+
+    gdf_lines = gpd.GeoDataFrame(
+        n.lines,
+        geometry=[
+            LineString(
+                [
+                    (n.buses.loc[line.bus0, "x"], n.buses.loc[line.bus0, "y"]),
+                    (n.buses.loc[line.bus1, "x"], n.buses.loc[line.bus1, "y"]),
+                ]
+            )
+            for line in n.lines.itertuples()
+        ],
+        crs=crs,
+    )
+
+    gdf_transformers = gpd.GeoDataFrame(
+        n.transformers,
+        geometry=[
+            LineString(
+                [
+                    (
+                        n.buses.loc[transformer.bus0, "x"],
+                        n.buses.loc[transformer.bus0, "y"],
+                    ),
+                    (
+                        n.buses.loc[transformer.bus1, "x"],
+                        n.buses.loc[transformer.bus1, "y"],
+                    ),
+                ]
+            )
+            for transformer in n.transformers.itertuples()
+        ],
+        crs=crs,
+    )
+
+    # Map related settings
+    bus_colors = mcolors.CSS4_COLORS["cadetblue"]
+    line_colors = mcolors.CSS4_COLORS["rosybrown"]
+    link_colors = mcolors.CSS4_COLORS["darkseagreen"]
+    transformer_colors = mcolors.CSS4_COLORS["orange"]
+
+    # Initialize the map
+    map = Map(tiles=None)
+
+    # Add tile layer legend entries
+    TileLayer("CartoDB dark_matter", name="CartoDB Dark Matter").add_to(map)
+    TileLayer("CartoDB positron", name="CartoDB Positron").add_to(map)
+    TileLayer("OpenStreetMap", name="OpenStreetMap").add_to(map)
+
+    components = ["buses", "lines", "links", "transformers"]
+    components_present = []
+
+    if not gdf_transformers.empty:
+        gdf_transformers.explore(
+            m=map,
+            color=transformer_colors,
+            tooltip=tooltip,
+            popup=popup,
+            name="Transformers",
+        )
+        components_present.append("transformers")
+
+    if not gdf_lines.empty:
+        gdf_lines.explore(
+            m=map, color=line_colors, tooltip=tooltip, popup=popup, name="Lines"
+        )
+        components_present.append("lines")
+
+    if not gdf_links.empty:
+        gdf_links.explore(
+            m=map, color=link_colors, tooltip=tooltip, popup=popup, name="Links"
+        )
+        components_present.append("links")
+
+    if not gdf_buses.empty:
+        buses_group = FeatureGroup(name="Buses")
```

Comment:
This is unused

Reply:
Thanks, relict of a feature I have not yet implemented :D